### DISCUSSION TITLE
How, when and where to use `import()` correctly

### DISCUSSION THREAD

**Discussion by SimJoSt** [deployphp/deployer#3654] [Help needed]
For more flexibility, we are switching some of our config files to PHP from YAML. It took us some time to get going, as we couldn't find an example PHP config in the docs. Using `dep init`, was the only way to get started :)

Instead of using `import:` in YAML `require` is used in PHP.
However, we also found the `import()` function in the API docs: https://deployer.org/docs/7.x/api#import
When should and how can it be used? Replacing `require` with `import()` didn't seem to work and isn't part of the `dep init` boilerplate.

**Comment by antonmedv** [comment]
The import() function is there to help to load yaml recipe in PHP recipe.

BTW, you can import PHP recipes in yaml as well. I use it sometimes: main recipe in yaml, but some tasks are defined in php.

**Reply by SimJoSt** [reply]
Thank you for the quick response 🙏 
Alright, so it's only use case it to load YAML recipes in PHP recipes. Not other PHP recipes or to use it in YAML. Got it.
I'll try that again and see, if we can get it working.

We tried importing our own PHP recipes in YAML and ran into some path issues, that you already commented on: https://github.com/deployphp/deployer/discussions/3653

_Originally commented on 27.07., but as it's own comment._
